implement batched serial pbtrf

yasahi-hpc opened the pull request:
This PR implements [pbtrf](https://netlib.org/lapack/explore-html/dc/d58/group__pbtrf_gafd2c3ba375ecee6dba1fba62696993e0.html) function.

Following files are added:
1. `KokkosBatched_Pbtrf_Serial_Impl.hpp`: Internal interfaces
2. `KokkosBatched_Pbtrf_Serial_Internal.hpp`: Implementation details
3. `KokkosBatched_Pbtrf.hpp`: APIs
4. `Test_Batched_SerialPbtrf.hpp`: Unit tests for that

## Detailed description
It computes the Cholesky factorization `U**H*U` (or `L*L**H`) factorization of a real symmetric (or complex Hermitian) positive definite band matrix `A`, where `A` is stored in a [band storage](https://www.netlib.org/lapack/lug/node124.html).
Here, the matrix has the following shape.
- `A`: `(batch_count, ldab, n)`  
  On entry, the upper or lower triangle of the symmetric band
  matrix A, stored in the first KD+1 rows of the array. On exit, the triangular factor `U` or `L` from the
  Cholesky factorization `A = U**H*U` or `A = L*L**H` of the band
  matrix `A`, in the same storage format as `A`

Example of a single batch of matrix A `n = 10`.

```bash
A
4 1 0 0 0 0 0 0 0 0 
1 4 1 0 0 0 0 0 0 0 
0 1 4 1 0 0 0 0 0 0 
0 0 1 4 1 0 0 0 0 0 
0 0 0 1 4 1 0 0 0 0 
0 0 0 0 1 4 1 0 0 0 
0 0 0 0 0 1 4 1 0 0 
0 0 0 0 0 0 1 4 1 0 
0 0 0 0 0 0 0 1 4 1 
0 0 0 0 0 0 0 0 1 4

AB (upper)
0 1 1 1 1 1 1 1 1 1 
4 4 4 4 4 4 4 4 4 4

AB (lower)
4 4 4 4 4 4 4 4 4 4
1 1 1 1 1 1 1 1 1 0
```

Parallelization would be made in the following manner. This is efficient only when 
A is given in `LayoutLeft` for GPUs and `LayoutRight` for CPUs (parallelized over batch direction).

```C++
Kokkos::parallel_for('pbtrf', 
    Kokkos::RangePolicy<execution_space> policy(0, n),
    [=](const int k) {
        auto aa = Kokkos::subview(_a, k, Kokkos::ALL(), Kokkos::ALL());

        KokkosBatched::SerialPbtrf<AlgoTagType>::invoke(aa);
    });
```

## Tests
1.  Make a real (complex Hermitian) symmetric positive definite banded matrix from random and keep it in a full (called `A`) or banded storage (called `AB`).
Then, factorize `AB` in banded storage to get `U` or `L` which is supposed to satisfy `A = L * L**H` (`A = U**H * U`).
Finally, check if  `A = L * L**H` (`A = U**H * U`).
1.  Simple and small analytical test, i.e. choose `A` as a known real (complex Hermitian) symmetric positive definite banded matrix, and do the same test.

kokkos-devops-admin commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           NO INSPECTION HAS BEEN PERFORMED ON THIS PULL REQUEST! - This PR must be inspected by setting label 'AT: PRE-TEST INSPECTED'.

lucbv reviewed (commented):
This looks fine overall, as you know I am a fan of having an analytical case if possible, maybe choleski on 1D FD matrix?

kokkos-devops-admin commented:
Status Flag 'Pre-Test Inspection' - SUCCESS: The last commit to this Pull Request has been INSPECTED by label AT: PRE-TEST INSPECTED! Autotester is Removing Label; this inspection will remain valid until a new commit to source branch is performed.

kokkos-devops-admin commented:
Status Flag 'Pull Request AutoTester' - Testing Jenkins Projects:

<details>
<summary>
  <b>Pull Request Auto Testing STARTING</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_CUDA11_CUDA11_LayoutRight</h3>
<ul>
  <li>Build Num: 1496</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>bbd6f1487a37d0f42d0c6e47ce3d6ec14ec12074</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_GCC930_Light_Tpls_GCC930_Tpls_CLANG13CUDA10</h3>
<ul>
  <li>Build Num: 1084</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>bbd6f1487a37d0f42d0c6e47ce3d6ec14ec12074</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_GNU1021</h3>
<ul>
  <li>Build Num: 738</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>bbd6f1487a37d0f42d0c6e47ce3d6ec14ec12074</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_GNU1021_Light_LayoutRight</h3>
<ul>
  <li>Build Num: 725</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>bbd6f1487a37d0f42d0c6e47ce3d6ec14ec12074</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_Tpls_GNU1021</h3>
<ul>
  <li>Build Num: 726</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>bbd6f1487a37d0f42d0c6e47ce3d6ec14ec12074</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_Tpls_INTEL19_solo</h3>
<ul>
  <li>Build Num: 730</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>bbd6f1487a37d0f42d0c6e47ce3d6ec14ec12074</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_CLANG1001_solo</h3>
<ul>
  <li>Build Num: 702</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>bbd6f1487a37d0f42d0c6e47ce3d6ec14ec12074</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_VEGA90A_ROCM561</h3>
<ul>
  <li>Build Num: 1175</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>bbd6f1487a37d0f42d0c6e47ce3d6ec14ec12074</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_VEGA90A_Tpls_ROCM561</h3>
<ul>
  <li>Build Num: 684</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>bbd6f1487a37d0f42d0c6e47ce3d6ec14ec12074</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Using Repos:</h2>

<dl>
  <dt>Repo: KOKKOSKERNELS (yasahi-hpc/kokkos-kernels)</dt>
  <ul>
    <li>Branch: implement-batched-serial-pbtrf</li>
    <li>SHA: bbd6f1487a37d0f42d0c6e47ce3d6ec14ec12074</li>
    <li>Mode: TEST_REPO</li>
  </ul>
</dl>

Pull Request Author: yasahi-hpc
</details>

kokkos-devops-admin commented:
Status Flag 'Pull Request AutoTester' - Jenkins Testing: all Jobs PASSED

<details>
<summary>
  <b>Pull Request Auto Testing has PASSED</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_CUDA11_CUDA11_LayoutRight</h3>
<ul>
  <li>Build Num: 1496</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>bbd6f1487a37d0f42d0c6e47ce3d6ec14ec12074</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_GCC930_Light_Tpls_GCC930_Tpls_CLANG13CUDA10</h3>
<ul>
  <li>Build Num: 1084</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>bbd6f1487a37d0f42d0c6e47ce3d6ec14ec12074</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_GNU1021</h3>
<ul>
  <li>Build Num: 738</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>bbd6f1487a37d0f42d0c6e47ce3d6ec14ec12074</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_GNU1021_Light_LayoutRight</h3>
<ul>
  <li>Build Num: 725</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>bbd6f1487a37d0f42d0c6e47ce3d6ec14ec12074</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_Tpls_GNU1021</h3>
<ul>
  <li>Build Num: 726</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>bbd6f1487a37d0f42d0c6e47ce3d6ec14ec12074</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_Tpls_INTEL19_solo</h3>
<ul>
  <li>Build Num: 730</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>bbd6f1487a37d0f42d0c6e47ce3d6ec14ec12074</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_CLANG1001_solo</h3>
<ul>
  <li>Build Num: 702</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>bbd6f1487a37d0f42d0c6e47ce3d6ec14ec12074</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_VEGA90A_ROCM561</h3>
<ul>
  <li>Build Num: 1175</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>bbd6f1487a37d0f42d0c6e47ce3d6ec14ec12074</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_VEGA90A_Tpls_ROCM561</h3>
<ul>
  <li>Build Num: 684</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>bbd6f1487a37d0f42d0c6e47ce3d6ec14ec12074</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>
</details>

kokkos-devops-admin commented:
Status Flag 'Pre-Merge Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           THE LAST COMMIT TO THIS PULL REQUEST HAS BEEN REVIEWED, BUT NOT ACCEPTED OR REQUIRES CHANGES!

kokkos-devops-admin commented:
All Jobs Finished; status = PASSED, However Inspection must be performed before merge can occur...

yasahi-hpc commented on batched/dense/impl/KokkosBatched_Pbtrf_Serial_Internal.hpp:
```diff
@@ -0,0 +1,268 @@
+    auto a_jj = AB[0 * as0 + j * as1].real();
+
+    // Check if L (j, j) is positive definite
+    if (a_jj <= 0) {
```
This is actually on purpose, based on the [reference implementation](https://www.netlib.org/lapack/explore-html/d4/d16/group__pbtf2_ga740a2a60e8a6cf6f397e6d4e2070df54.html#ga740a2a60e8a6cf6f397e6d4e2070df54). After some consideration, I added a debug level guard. In case `L (j, j)` is not positive definite, subsequent operations are invalid anyway. We do not care whether the complex part of `L (j, j)` is suppressed or not for the invalid case.

yasahi-hpc commented:
> This looks fine overall, as you know I am a fan of having an analytical case if possible, maybe choleski on 1D FD matrix?

Thank you for reviews. I have fixed based on your comments. I have also added a pure analytical test case for 2x2 matrix.

kokkos-devops-admin commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           NO INSPECTION HAS BEEN PERFORMED ON THIS PULL REQUEST! - This PR must be inspected by setting label 'AT: PRE-TEST INSPECTED'.

lucbv reviewed (approved):
Looks good to me, thanks @yasahi-hpc

lucbv commented:
@ndellingwood: as you can see here, the AT-2 is running correctly after review, I am not sure why it is not the case for the release PR?

kokkos-devops-admin commented:
Status Flag 'Pre-Test Inspection' - SUCCESS: The last commit to this Pull Request has been INSPECTED by label AT: PRE-TEST INSPECTED! Autotester is Removing Label; this inspection will remain valid until a new commit to source branch is performed.

kokkos-devops-admin commented:
Status Flag 'Pull Request AutoTester' - Testing Jenkins Projects:

<details>
<summary>
  <b>Pull Request Auto Testing STARTING</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_CUDA11_CUDA11_LayoutRight</h3>
<ul>
  <li>Build Num: 1498</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>03341692b3b86d128e77410474a5cad0441008b3</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_GCC930_Light_Tpls_GCC930_Tpls_CLANG13CUDA10</h3>
<ul>
  <li>Build Num: 1086</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>03341692b3b86d128e77410474a5cad0441008b3</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_GNU1021</h3>
<ul>
  <li>Build Num: 740</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>03341692b3b86d128e77410474a5cad0441008b3</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_GNU1021_Light_LayoutRight</h3>
<ul>
  <li>Build Num: 727</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>03341692b3b86d128e77410474a5cad0441008b3</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_Tpls_GNU1021</h3>
<ul>
  <li>Build Num: 728</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>03341692b3b86d128e77410474a5cad0441008b3</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_Tpls_INTEL19_solo</h3>
<ul>
  <li>Build Num: 732</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>03341692b3b86d128e77410474a5cad0441008b3</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_CLANG1001_solo</h3>
<ul>
  <li>Build Num: 704</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>03341692b3b86d128e77410474a5cad0441008b3</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_VEGA90A_ROCM561</h3>
<ul>
  <li>Build Num: 1177</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>03341692b3b86d128e77410474a5cad0441008b3</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_VEGA90A_Tpls_ROCM561</h3>
<ul>
  <li>Build Num: 685</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>03341692b3b86d128e77410474a5cad0441008b3</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Using Repos:</h2>

<dl>
  <dt>Repo: KOKKOSKERNELS (yasahi-hpc/kokkos-kernels)</dt>
  <ul>
    <li>Branch: implement-batched-serial-pbtrf</li>
    <li>SHA: 03341692b3b86d128e77410474a5cad0441008b3</li>
    <li>Mode: TEST_REPO</li>
  </ul>
</dl>

Pull Request Author: yasahi-hpc
</details>

kokkos-devops-admin commented:
Status Flag 'Pull Request AutoTester' - Jenkins Testing: all Jobs PASSED

<details>
<summary>
  <b>Pull Request Auto Testing has PASSED</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_CUDA11_CUDA11_LayoutRight</h3>
<ul>
  <li>Build Num: 1498</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>03341692b3b86d128e77410474a5cad0441008b3</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_GCC930_Light_Tpls_GCC930_Tpls_CLANG13CUDA10</h3>
<ul>
  <li>Build Num: 1086</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>03341692b3b86d128e77410474a5cad0441008b3</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_GNU1021</h3>
<ul>
  <li>Build Num: 740</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>03341692b3b86d128e77410474a5cad0441008b3</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_GNU1021_Light_LayoutRight</h3>
<ul>
  <li>Build Num: 727</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>03341692b3b86d128e77410474a5cad0441008b3</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_Tpls_GNU1021</h3>
<ul>
  <li>Build Num: 728</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>03341692b3b86d128e77410474a5cad0441008b3</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_Tpls_INTEL19_solo</h3>
<ul>
  <li>Build Num: 732</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>03341692b3b86d128e77410474a5cad0441008b3</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_CLANG1001_solo</h3>
<ul>
  <li>Build Num: 704</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>03341692b3b86d128e77410474a5cad0441008b3</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_VEGA90A_ROCM561</h3>
<ul>
  <li>Build Num: 1177</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>03341692b3b86d128e77410474a5cad0441008b3</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: KokkosKernels_PullRequest_VEGA90A_Tpls_ROCM561</h3>
<ul>
  <li>Build Num: 685</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_REPO</td>
    <td>https://github.com/yasahi-hpc/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_SOURCE_SHA</td>
    <td>03341692b3b86d128e77410474a5cad0441008b3</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_REPO</td>
    <td>https://github.com/kokkos/kokkos-kernels</td>
  </tr>

  <tr>
    <td>KOKKOSKERNELS_TARGET_SHA</td>
    <td>227349e87c0c792ef6036d8af0d85d9040de1fb9</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>feature request</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2322</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>KOKKOSKERNELS</td>
  </tr>
</table>
</details>

kokkos-devops-admin commented:
Status Flag 'Pre-Merge Inspection' - SUCCESS: The last commit to this Pull Request has been INSPECTED AND APPROVED by [ lucbv ]!

kokkos-devops-admin commented:
Status Flag 'Pull Request AutoTester' - Pull Request MUST BE MERGED MANUALLY BY Project Team - This Repo does not support Automerge

kokkos-devops-admin commented:
Status Flag 'Pull Request AutoTester' - Pull Request MUST BE MERGED MANUALLY BY Project Team - This Repo does not support Automerge
